src/main/java/com/redhat/sast/api/service/DashboardService.java

```diff
@@ -0,0 +1,126 @@
+    private final OshUncollectedScanRepository oshUncollectedScanRepository;
+
+    private final AtomicReference<CachedSummary> cachedSummary = new AtomicReference<>();
+
+    /**
+     * Gets dashboard summary statistics with caching.
+     *
+     * @return aggregated dashboard statistics
+     */
+    public DashboardSummaryDto getSummary() {
+        CachedSummary cached = cachedSummary.get();
+
+        if (cached != null && !cached.isExpired()) {
+            LOGGER.debug("Returning cached dashboard summary (age: {} seconds)", cached.getAgeSeconds());
+            return cached.summary;
+        }
+
+        DashboardSummaryDto summary = calculateSummary();
+
+        cachedSummary.set(new CachedSummary(summary, LocalDateTime.now()));
+
+        LOGGER.debug("Calculated and cached new dashboard summary");
+        return summary;
+    }
+
+    /**
+     * Invalidates the cache, forcing next request to recalculate.
+     */
+    public void invalidateCache() {
+        cachedSummary.set(null);
+        LOGGER.debug("Dashboard summary cache invalidated");
+    }
+
+    /**
+     * Calculates dashboard summary by executing database queries.
+     */
+    private DashboardSummaryDto calculateSummary() {
+        DashboardSummaryDto dto = new DashboardSummaryDto();
+
+        // Job statistics
+        dto.setTotalJobs(jobRepository.count());
+        dto.setPendingJobs(jobRepository.count(STATUS_FIELD, JobStatus.PENDING));
+        dto.setRunningJobs(jobRepository.count(STATUS_FIELD, JobStatus.RUNNING));
+        dto.setCompletedJobs(jobRepository.count(STATUS_FIELD, JobStatus.COMPLETED));
+        dto.setFailedJobs(jobRepository.count(STATUS_FIELD, JobStatus.FAILED));
+        dto.setCancelledJobs(jobRepository.count(STATUS_FIELD, JobStatus.CANCELLED));
+
+        // Batch statistics
+        dto.setTotalBatches(jobBatchRepository.count());
+        dto.setProcessingBatches(jobBatchRepository.count(STATUS_FIELD, BatchStatus.PROCESSING));
+        dto.setCompletedBatches(jobBatchRepository.count(STATUS_FIELD, BatchStatus.COMPLETED));
+
+        // OSH scan statistics
+        long collectedOshScans = countJobsWithOshScanId();
+        long uncollectedOshScans = oshUncollectedScanRepository.count();
+
+        dto.setCollectedOshScans(collectedOshScans);
+        dto.setUncollectedOshScans(uncollectedOshScans);
+        dto.setTotalOshScans(collectedOshScans + uncollectedOshScans);
+
+        dto.setTimestamp(LocalDateTime.now());
+
+        return dto;
+    }
+
+    /**
+     * Counts jobs that have an associated OSH scan ID (collected scans).
+     */
+    private long countJobsWithOshScanId() {
+        return jobRepository
+                .getEntityManager()
+                .createQuery("SELECT COUNT(j) FROM Job j WHERE j.oshScanId IS NOT NULL", Long.class)
+                .getSingleResult();
+    }
+
+    /**
+     * Internal class for cache entry with expiration.
+     */
+    private static class CachedSummary {
```

Comment:
can be defined as a record. Please move it to the top of the class after the variables section.